clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -641,65 +641,66 @@ void SYCL::gen::BackendCompiler::ConstructJob(Compilation &C,
-               .Cases("intel_gpu_cfl", "intel_gpu_9_2_9", "cfl")
-               .Cases("intel_gpu_apl", "intel_gpu_9_3_0", "apl")
-               .Cases("intel_gpu_glk", "intel_gpu_9_4_0", "glk")
-               .Cases("intel_gpu_whl", "intel_gpu_9_5_0", "whl")
-               .Cases("intel_gpu_aml", "intel_gpu_9_6_0", "aml")
-               .Cases("intel_gpu_cml", "intel_gpu_9_7_0", "cml")
-               .Cases("intel_gpu_icllp", "intel_gpu_11_0_0", "icllp")
-               .Cases("intel_gpu_tgllp", "intel_gpu_12_0_0", "tgllp")
-               .Case("intel_gpu_rkl", "rkl")
-               .Case("intel_gpu_adl_s", "adl_s")
-               .Case("intel_gpu_rpl_s", "rpl_s")
-               .Case("intel_gpu_adl_p", "adl_p")
-               .Case("intel_gpu_adl_n", "adl_n")
-               .Cases("intel_gpu_dg1", "intel_gpu_12_10_0", "dg1")
-               .Case("intel_gpu_acm_g10", "acm_g10")
-               .Case("intel_gpu_acm_g11", "acm_g11")
-               .Case("intel_gpu_acm_g12", "acm_g12")
-               .Case("intel_gpu_pvc", "pvc")
-               .Case("nvidia_gpu_sm_50", "sm_50")
-               .Case("nvidia_gpu_sm_52", "sm_52")
-               .Case("nvidia_gpu_sm_53", "sm_53")
-               .Case("nvidia_gpu_sm_60", "sm_60")
-               .Case("nvidia_gpu_sm_61", "sm_61")
-               .Case("nvidia_gpu_sm_62", "sm_62")
-               .Case("nvidia_gpu_sm_70", "sm_70")
-               .Case("nvidia_gpu_sm_72", "sm_72")
-               .Case("nvidia_gpu_sm_75", "sm_75")
-               .Case("nvidia_gpu_sm_80", "sm_80")
-               .Case("nvidia_gpu_sm_86", "sm_86")
-               .Case("nvidia_gpu_sm_87", "sm_87")
-               .Case("nvidia_gpu_sm_89", "sm_89")
-               .Case("nvidia_gpu_sm_90", "sm_90")
-               .Case("amd_gpu_gfx700", "gfx700")
-               .Case("amd_gpu_gfx701", "gfx701")
-               .Case("amd_gpu_gfx702", "gfx702")
-               .Case("amd_gpu_gfx801", "gfx801")
-               .Case("amd_gpu_gfx802", "gfx802")
-               .Case("amd_gpu_gfx803", "gfx803")
-               .Case("amd_gpu_gfx805", "gfx805")
-               .Case("amd_gpu_gfx810", "gfx810")
-               .Case("amd_gpu_gfx900", "gfx900")
-               .Case("amd_gpu_gfx902", "gfx902")
-               .Case("amd_gpu_gfx904", "gfx904")
-               .Case("amd_gpu_gfx906", "gfx906")
-               .Case("amd_gpu_gfx908", "gfx908")
-               .Case("amd_gpu_gfx90a", "gfx90a")
-               .Case("amd_gpu_gfx1010", "gfx1010")
-               .Case("amd_gpu_gfx1011", "gfx1011")
-               .Case("amd_gpu_gfx1012", "gfx1012")
-               .Case("amd_gpu_gfx1013", "gfx1013")
-               .Case("amd_gpu_gfx1030", "gfx1030")
-               .Case("amd_gpu_gfx1031", "gfx1031")
-               .Case("amd_gpu_gfx1032", "gfx1032")
-               .Case("amd_gpu_gfx1034", "gfx1034")
-               .Default("");
+  Device =
+      llvm::StringSwitch<StringRef>(DeviceName)
```

Comment:
Note for reviewers: clang-format insisted on this ugly change, so adding below what was changed in this part of the file:

```C++
--- .Cases("intel_gpu_apl", "intel_gpu_9_3_0", "apl")
+++ .Cases("intel_gpu_apl", "intel_gpu_bxt", "intel_gpu_9_3_0", "apl")

...

+++ .Cases("intel_gpu_ehl", "intel_gpu_jsl", "ehl")

...

--- .Case("intel_gpu_adl_s", "adl_s")
--- .Case("intel_gpu_rpl_s", "rpl_s")
+++ .Cases("intel_gpu_adl_s", "intel_gpu_rpl_s", "adl_s")

...

--- .Case("intel_gpu_acm_g10", "acm_g10")
+++ .Cases("intel_gpu_acm_g10", "intel_gpu_dg2_g10", "acm_g10")
--- .Case("intel_gpu_acm_g11", "acm_g11")
+++ .Cases("intel_gpu_acm_g11", "intel_gpu_dg2_g11", "acm_g11")
--- .Case("intel_gpu_acm_g12", "acm_g12")
+++ .Cases("intel_gpu_acm_g12", "intel_gpu_dg2_g12", "acm_g12")
```

Reply:
May be it makes sense to add a "clang-format off" comment here? We are unlikely to need this, but it will save the history of changes.

Reply:
Either option LGTM, @mdtoguchi what do you think, should we add "clang-format off"?

Reply:
I'm OK with leaving the formatting changes in.